include/onnx-mlir/Runtime/OMTensor.h

```diff
@@ -0,0 +1,387 @@
+ * partially filled omTensor; prefer using the new omTensorCreateEmpty()
+ * function to fill shape & stride fields automatically.
+ *
+ * @param rank tensor rank
+ * @return pointer to OMTensor created, NULL if creation failed.
+ *
+ */
+OMTensor *omTensorCreateEmptyDeprecated(int rank);
+
+/**
+ * Destroy the OMTensor struct.
+ *
+ * @param tensor pointer to the OMTensor
+ *
+ */
+void omTensorDestroy(OMTensor *tensor);
+
+/**
+ * OMTensor aligned data pointer getter
+ *
+ * @param tensor pointer to the OMTensor
+ * @return pointer to the data buffer of the OMTensor,
+ *         NULL if the data buffer is not set.
+ */
+void *omTensorGetAlignedPtr(OMTensor *tensor);
+
+/**
+ * OMTensor allocated and aligned pointer setter.
+ *
+ * @param tensor pointer to the OMTensor
+ * @param owning whether allocatedPtr should be freed after tensor is destroyed.
+ * @param allocatedPtr allocated pointer to tensor content.
+ * @param alignedPtr aligned pointer to tensor content. If NULL will be set to
+ * allocatedPtr.
+ *
+ *
+ */
+void omTensorSetPtr(
+    OMTensor *tensor, int owning, void *allocatedPtr, void *alignedPtr);
+
+/**
+ * OMTensor allocated ptr getter.
+ *
+ * @param tensor pointer to the OMTensor
+ * @return pointer to the allocated data buffer of the OMTensor,
+ *         NULL if the allocated data buffer is not set.
+ */
+void *omTensorGetAllocatedPtr(OMTensor *tensor);
+
+/**
+ * OMTensor data sizes getter
+ *
+ * @param tensor pointer to the OMTensor
+ * @return pointer to the data shape array.
+ */
+int64_t *omTensorGetDataShape(OMTensor *tensor);
+
+/**
+ * OMTensor data sizes setter
+ *
+ * @param tensor pointer to the OMTensor
+ * @param shape data sizes array to be set
+ *
+ * Set the data sizes array of the OMTensor to the values in the input array.
+ */
+void omTensorSetShape(OMTensor *tensor, int64_t *shape);
+
+/**
+ * OMTensor data strides getter
+ *
+ * @param tensor pointer to the OMTensor
+ * @return pointer to the data strides array.
+ */
+int64_t *omTensorGetStrides(OMTensor *tensor);
+
+/**
+ * OMTensor data strides setter
+ *
+ * @param tensor pointer to the OMTensor
+ * @param strides tensor strides array to be set.
+ *
+ * Set the data strides array of the OMTensor to the values in the input array.
+ */
+void omTensorSetStrides(OMTensor *tensor, int64_t *strides);
+
+/**
+ * OMTensor data type getter
+ *
+ * @param tensor pointer to the OMTensor
+ * @return ONNX data type of the data buffer elements.
+ */
+OM_DATA_TYPE omTensorGetDataType(OMTensor *tensor);
+
+/**
+ * OMTensor data type setter
+ *
+ * @param tensor pointer to the OMTensor
+ * @param dataType ONNX data type to be set
+ *
+ * Set the ONNX data type of the data buffer elements.
+ */
+void omTensorSetDataType(OMTensor *tensor, OM_DATA_TYPE dataType);
+
+/* Helper function to get the ONNX data type size in bytes */
+static inline int getDataTypeSize(OM_DATA_TYPE dataType) {
+  return OM_DATA_TYPE_SIZE[dataType];
+}
+
+/**
```

Comment:
It seems somewhat obvious to me that the logical size is the one that makes sense here since we are calculating the number of elements in the tensor, not buffer size. The only thing that could go wrong is when you have gaps in the array (in the form of non-trivial strides), and I explicitly ensure that strides contain no gap.